Tools/AP_Periph/can.cpp

```diff
@@ -1950,6 +1958,16 @@ void AP_Periph_FW::can_update()
 
         }
     }
+
+    // if there is a reboot scheduled, do it 1 second after request to allow the acknowledgement to be sent
+    if (reboot_request_ms != 0 && (AP_HAL::millis() - reboot_request_ms > 1000)) {
+        prepare_reboot();
+#if CONFIG_HAL_BOARD == HAL_BOARD_CHIBIOS
+        NVIC_SystemReset();
+#elif CONFIG_HAL_BOARD == HAL_BOARD_SITL
+        HAL_SITL::actually_reboot();
+#endif
```

Comment:
pre-existing, and doesn't need fixing here, but seems to say we should have an `actually_reboot` method on the HAL.  Preferably with a better name.